torch/optim/optimizer.py

```diff
@@ -491,8 +590,84 @@ def _process_value_according_to_param_policy(
             else:
                 return value.to(device=param.device)
 
+
+    def register_load_state_dict_pre_hook(
+        self,
+        hook: Callable[["Optimizer", StateDict], Optional[StateDict]],
+        prepend: bool = False,
+    ) -> RemovableHandle:
+        r"""Register a load_state_dict pre-hook which will be called before
+        :meth:`~torch.optim.Optimizer.load_state_dict` is called. It should have the
+        following signature::
+
+            hook(optimizer, state_dict) -> None or state_dict
```

Comment:
```suggestion
            hook(optimizer, state_dict) -> state_dict or None
```
For consistency with the other one!